synfig-studio/src/synfigapp/settings.cpp

```diff
@@ -278,3 +278,79 @@ Settings::load_from_file(const synfig::String& filename, const synfig::String& k
 		return false;
 	return true;
 }
+
+template<> int
+Settings::get_value<int>(const synfig::String& key, const int& default_value) const {
```

Comment:
> But I think it makes easy to implement custom versions if you want to ("template specialization") without changing the class.


Yes. But it also leads to strange errors where the code compiles fine, but you get an error at link time. 

For example:
```
    struct T {};
    T default_val;
    _preferences.get_value("test", default_val);
```

Error:
```
/usr/bin/ld: CMakeFiles/synfigstudio.dir/app.cpp.o: in function `studio::App::dialog_open_file_sketch(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)':
synfig/synfig-studio/src/gui/app.cpp:2576: undefined reference to `T synfigapp::Settings::get_value<T, (void*)0>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, T const&) const'
```

In case of non-templated version:
```
error: no matching function for call to ‘Preferences::get_value(std::string, T&)’
```

This is an explicit type definition, but sometimes you can get an implicit conversion and waste a lot of time trying to figure out what's going on. 

Anyway, the final decision is up to you, so please rebase when you think the code is ready.

Reply:
Changed and rebased.